server/aws-lsp-codewhisperer/src/language-server/utils.ts

```diff
@@ -131,3 +133,15 @@ export const getUserAgent = (initializeParams: InitializeParams, serverInfo?: Se
 
     return items.join(' ')
 }
+
+export function getLoginTypeFromProvider(credentialsProvider: CredentialsProvider): LoginType {
+    if (credentialsProvider.hasCredentials('iam')) {
+        return 'iam'
+    }
+    const connectionMetadata = credentialsProvider.getConnectionMetadata()
+    return connectionMetadata && connectionMetadata.sso && connectionMetadata.sso.startUrl
+        ? connectionMetadata.sso.startUrl.includes(builderIdStartUrl)
+            ? 'builderId'
+            : 'identityCenter'
+        : 'builderId'
```

Comment:
Why this falls back to `builderId`, is this fallback scenario in VSCode as well?

Reply:
yes..if I see this code; they have a comment that it must be IDC by the process of elimination..in vscode they try to check using some connection parameters but here we have builderId URL defined but idc start urls can be different - that's why this condition.
`    public isEnterpriseSsoInUse(): boolean {
        const conn = this.conn
        // we have an sso that isn't builder id, must be IdC by process of elimination
        const isUsingEnterpriseSso = conn?.type === 'sso' && !isBuilderIdConnection(conn)
        return conn !== undefined && isUsingEnterpriseSso
    }`